src/main/kotlin/com/exactpro/th2/act/ActHandlerKt.kt

```diff
@@ -0,0 +1,390 @@
+            responseObserver.sendErrorResponse("Failed to place a QuoteResponse. Error: ${it.message}")
+        }.onSuccess {
+            LOGGER.debug { "placeQuoteResponseFIX successfully finished" }
+        }
+    }
+
+    override fun placeQuoteFIX(
+        request: PlaceMessageRequest,
+        responseObserver: StreamObserver<PlaceMessageResponse>
+    ) {
+        runCatching {
+            LOGGER.debug { "placeQuoteFIX request: $request" }
+            val expectedValue = requireNotNull(request.message.fieldsMap["RFQID"])
+            val settings = CallSettings("placeQuoteFIX", expectedValue.simpleValue, mapOf("QuoteAck" to "RFQID"))
+            placeMessage(settings, request, responseObserver)
+        }.onFailure {
+            LOGGER.error(it) { "Failed to place an Quote. Message = ${request.message}" }
+            responseObserver.sendErrorResponse("Failed to place a Quote. Error: ${it.message}")
+        }.onSuccess {
+            LOGGER.debug { "placeQuoteFIX successfully finished" }
+        }
+    }
+
+    private fun placeMessage(settings: CallSettings, request: PlaceMessageRequest, responseObserver: StreamObserver<PlaceMessageResponse>) {
+        LOGGER.debug { "Begin place ${settings.name} ${shortDebugString(request)}" }
+        runCatching {
+            val startTime = Instant.now()
+            val actEventId = createAndStoreParentEvent(request, settings.name, PASSED)
+
+            runCatching {
+                val message: Message = backwardCompatibilityConnectionId(request)
+                val requestConnId = message.metadata.id.connectionId
+
+                val checkpoint = registerCheckPoint(actEventId)
+                val monitor = MessageResponseMonitor()
+
+                createMessageReceiver(settings.expectedValue, settings.expectedFields, monitor, requestConnId).use { messageReceiver ->
+                    sendMessages(settings.name, actEventId, startTime, message)
+                    val timeout: Long = getTimeout(Context.current().deadline)
+                    monitor.awaitSync(timeout, TimeUnit.MILLISECONDS)
+                    messageReceiver.processResponseMessage(settings.name, checkpoint, actEventId, responseObserver, timeout)
+                }
+            }.onFailure {
+                eventRouter.storeErrorEvent(settings.name, startTime, actEventId, it)
+                throw it
+            }
+
+        }.onFailure {
+            LOGGER.error(it) { "Failed to place ${settings.name}" }
+            responseObserver.sendErrorResponse("Failed to place ${settings.name}, error: ${it.message}")
+        }
+        LOGGER.debug { "End place ${settings.name}" }
+    }
+
+    private fun AbstractMessageReceiver.processResponseMessage(
+        callName: String,
+        checkpoint: Checkpoint,
+        parentEventId: EventID,
+        responseObserver: StreamObserver<PlaceMessageResponse>,
+        timeout: Long,
+    ) {
+        responseMessage?.let { response ->
```

Comment:
Looks like we never call `responseObserver.onNext + responseObserver.onCompleted` on `responseObserver.onError` in case the `responseMessage` is `null`. It will cause the response will never be sent and the connection will be active until the user breaks it